### PR TITLE
Add built-in Sidekiq latency check

### PR DESCRIPTION
**Pull request by dwbutler**
Hi, thanks for building OkComputer!

I needed a Sidekiq queue check in my application. Sidekiq's recommended way to monitor a queue is to check for latency (the age of the oldest job).

This was totally based off of the backed-up Resque queue check.

I figured since Sidekiq is pretty widely used, it would be nice to include this as a built-in check.

### CODE REVIEW

File: lib/ok_computer/built_in_checks/sidekiq_latency_check.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+module OkComputer
+  class SidekiqLatencyCheck < SizeThresholdCheck
+    attr_accessor :queue
+
+    # Public: Initialize a check for a backed-up Sidekiq queue
+    # See https://github.com/mperham/sidekiq/wiki/Monitoring#monitoring-queue-latency
+      #
+      # queue - The name of the Sidekiq queue to check
+      # threshold - An Integer to compare the queue's latency against to consider
```
Comment (pbyrne) [inline lib/ok_computer/built_in_checks/sidekiq_latency_check.rb]: A couple of pedantic notes:

* The comments here are out of alignment, with some indented 2 more spaces than the others.
* Could you update `threshold` on this line to match the new name, `latency` in the method definition itself?

**Comment by pbyrne** [comment]
I left a note with a couple quibbles, but this change looks good to me. Once you clean those up, I think it's good to merge.

As the number of built-in checks grows, it begins to seem that the tool-specific checks probably belong in some kind of `okcomputer-contrib` gem separate from the core `okcomputer` Rails engine. But that's a problem for another day and doesn't block this PR.

File: lib/ok_computer/built_in_checks/sidekiq_latency_check.rb (hunk 1 above)
Comment (dwbutler) [inline lib/ok_computer/built_in_checks/sidekiq_latency_check.rb]: Good catch. I decided that naming it `threshold` would be a better description for what the variable does.

**Comment by dwbutler** [comment]
Thanks, I cleaned up the formatting and variable names. (I decided that naming the parameter `threshold` was probably more descriptive.)

I was actually thinking the exact same thing about a `okcomputer-contrib` gem. I've actually written an elasticsearch check that I need to reuse in several different apps. I was going to ask if I should release it separately. If you want to push a basic skeleton for `okcomputer-contrib`, I can make that my first contribution!

**Comment by pbyrne** [comment]
> Thanks, I cleaned up the formatting and variable names.

Perfect. Merging shortly, and I'll cut a new release.

> I've actually written an elasticsearch check that I need to reuse in several different apps. I was going to ask if I should release it separately. If you want to push a basic skeleton for okcomputer-contrib, I can make that my first contribution!

I like your energy!

I don't know that we're 100% going to pull the trigger on `okcomputer-contrib`, so I don't want that to hold you back from contributing your elasticsearch check right away. I think I'd be fine with either of these two scenarios:

* You contribute it to this repo as another PR and we merge it, and then later we can extract it into `okcomputer-contrib` along with the others.
* You can release it as a stand-alone gem (a la the now-defunct [okcomputer-newrelic-ignore](https://rubygems.org/gems/okcomputer-newrelic-ignore) gem) and we can link to it from the README. Then when we make `okcomputer-contrib` you can submit it there.

**Comment by dwbutler** [comment]
Cool, I'll submit another pull request then, and leave it to your discretion if/when to extract things out to a separate gem.

**Pull request merged**